src/main/java/org/terasology/module/behaviors/actions/FindPathToNode.java

```diff
@@ -2,95 +2,87 @@
-        actor.save(moveComponent);
-        WalkableBlock currentBlock = moveComponent.currentBlock;
-        if (currentBlock == null || targetLocation == null) {
-            moveComponent.path = Path.INVALID;
-            return;
+        if (pluginSystem == null) {
+            pluginSystem = CoreRegistry.get(PluginSystem.class);
         }
-        WalkableBlock workTarget = navGraphSystem.getBlock(targetLocation);
-        if (workTarget == null) {
-            moveComponent.path = Path.INVALID;
-            return;
-        }
-        SettableFuture<List<Path>> pathFuture = pathfinderSystem.requestPath(
-                actor.getEntity(), currentBlock.getBlockPosition(),
-                Arrays.asList(workTarget.getBlockPosition()));
 
-        Futures.addCallback(pathFuture, new FutureCallback<List<Path>>() {
-            @Override
-            public void onSuccess(List<Path> paths) {
-                if (paths == null) {
-                    moveComponent.path = Path.INVALID;
-                } else if (paths.size() > 0) {
-                    moveComponent.path = paths.get(0);
-                }
-                actor.save(moveComponent);
-            }
+        MinionMoveComponent flexibleMovementComponent = actor.getComponent(MinionMoveComponent.class);
+        flexibleMovementComponent.running = true;
+        Vector3ic start = Blocks.toBlockPos(actor.getComponent(LocationComponent.class).getWorldPosition(new Vector3f()));
+        Vector3ic goal = actor.getComponent(MinionMoveComponent.class).getPathGoal();
 
-            @Override
-            public void onFailure(Throwable t) {
-                moveComponent.path = Path.INVALID;
+        JPSConfig config = new JPSConfig(start, goal);
+        config.useLineOfSight = false;
+        config.requester = actor.getEntity();
+        config.maxTime = 10f;
+        config.maxDepth = 150;
+        config.goalDistance = flexibleMovementComponent.pathGoalDistance;
+        config.plugin = pluginSystem.getMovementPlugin(actor.getEntity()).getJpsPlugin(actor.getEntity());
+
+        int id = pathfinderSystem.requestPath(config, (path, target) -> {
+            MinionMoveComponent flexibleMovementComponent1 = actor.getComponent(MinionMoveComponent.class);
+            flexibleMovementComponent1.running = false;
+            if (path == null || path.size() == 0) {
+                actor.save(flexibleMovementComponent1);
+                return;
             }
-        }, MoreExecutors.directExecutor());
+            path.remove(0);
+
+            flexibleMovementComponent1.setPath(path);
+
+            actor.save(flexibleMovementComponent1);
+        });
     }
 
     @Override
     public BehaviorState modify(Actor actor, BehaviorState result) {
-        final MinionMoveComponent moveComponent = actor.getComponent(MinionMoveComponent.class);
-        if (moveComponent.path == null) {
+        // this an action node, so it will always be called with BehaviorState.UNDEFINED
```

Comment:
This part is true, though - I'd like to keep this comment as help for further work on this.